kokoro/scripts/build/build_package.sh

```diff
@@ -43,41 +43,30 @@ export_to_sponge_config "PACKAGE_VERSION" "${PKG_VERSION}"
 ARCH="$(docker info --format '{{.Architecture}}')"
 ARTIFACT_REGISTRY="us-docker.pkg.dev"
 docker-credential-gcr configure-docker --registries="${ARTIFACT_REGISTRY}"
-CACHE_LOCATION="${ARTIFACT_REGISTRY}/stackdriver-test-143416/google-cloud-ops-agent-build-cache/ops-agent-cache:${DISTRO}_${ARCH}"
+CACHE="${ARTIFACT_REGISTRY}/stackdriver-test-143416/google-cloud-ops-agent-build-cache/ops-agent-cache"
 
-DOCKER_BUILDKIT=1 docker build . \
-  --cache-from="${CACHE_LOCATION}" \
+build_params=()
+if [[ -n "${KOKORO_GITHUB_PULL_REQUEST_NUMBER}" ]]; then  # Per-PR cache
+    build_params+=(--cache-from=type=registry,ref="${CACHE}:${DISTRO}_${ARCH}_PR${KOKORO_GITHUB_PULL_REQUEST_NUMBER}")
```

Comment:
this means that the first build for any pull request will be uncached, right?

Reply:
There should always be the base cache. If the build runs from a PR, we'll try to cache from two caches.